ADR/0047-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,220 @@
+| External Secrets Operator (ESO) | Delivery of secrets to Kubernetes |
+| Konflux UI | Monitoring and runtime insight only |
+| Fleet Registry | Service discovery and instance metadata for multi-instance deployments |
+
+### VS Code Plugin Design
+
+**Core Features:**
+- Forms and wizards to scaffold and edit YAML objects
+- Built-in schema-aware YAML editor
+- Linting and validation against Konflux schemas
+- Integration with local secret store session (Vault or Kubernetes)
+- Git-aware diff view and PR-ready commit generation
+- Optional: preview pipeline graph / dependency resolution
+
+**Secrets UX:**
+- Form-based secret creation (with validation, TTLs, metadata)
+- Generated secret definitions point to secret store paths via ESO
+- Secrets created in the secret store configured for that instance of Konflux
+- Initially supported stores: Vault and Kubernetes Secrets
+- Optional: audit log stored in `.konflux/` for local tracking
+
+**GitOps Flow:**
+1. Developer opens repo in VS Code
+2. Launches Konflux plugin
+3. Uses wizard to configure new component/build
+4. Adds secrets as needed via plugin form
+5. Plugin generates YAML and commits to Git
+6. Developer pushes PR
+7. CI runs lightweight schema validation (should pass if plugin was used)
+
+### Multi-Repo GitOps Model
+
+**Tenant Onboarding via Repo Registration:**
+- Konflux will support multiple GitOps repositories, each representing an independent tenant
+- Each registered GitOps repo is treated as a distinct tenant with its own Kubernetes namespace
+- Teams register GitOps repo via VS Code plugin, CLI (`konflux register`), or UI
+- Konflux provisions a new Kubernetes namespace and runs initial validation
+- Decouples team onboarding and eliminates complex CODEOWNERS management
+
+### Secrets Management Integration
+
+**Core Capability:**
+- VS Code plugin enables declarative secrets creation directly from the IDE
+- Secrets are managed through Git configuration with secure backend integration
+- Generated secret definitions use External Secrets Operator (ESO) to reference backend stores
+
+**Supported Backends:**
+- **Kubernetes Secrets**: Native Kubernetes secret storage
+- **Vault**: HashiCorp Vault integration for enhanced security features
+- Future backends can be added as needed
+
+**Security Principles:**
+- Plugin only creates secrets, does not read or list existing secrets
+- All secret operations are scoped and audited
+- Optional: CLI fallback (`konflux secrets create`)
+
+### Fleet Awareness
+
+**Multi-Instance Discovery:**
+When a user's organization operates multiple Konflux instances, the VS Code plugin should provide fleet-wide awareness and intelligent instance selection. Fleet metadata is simple metadata with a description of the clusters capabilities and limitations (text), plus an indicator if the cluster has capacity for new tenants namespace. Each Konflux instance should have a complete copy of the fleet metadata.
```

Comment:
This seems like it is its own idea. Break this out into a separate ADR to flesh it out.

What's the metadata look like? How does a cluster know if its full? How does each instance get a copy of the fleet metadata? How do instances join the fleet? What defines the fleet? If two instances have different copies of the fleet metadata, how does that get reconciled? Can you get the fleet metadata without otherwise being authorized on a cluster? If not how do you get authorized in the first place?

Reply:
Besides - this feels like a _generic_ problem. What other fleet management technologies should we look at before building anything new here?

Reply:
I second that fleet management should be its own ADR. I would even go as far and question if we need any idea of a fleet on the cluster level.